languagetool-dev/src/main/java/org/languagetool/dev/httpchecker/HttpApiSentenceChecker.java

```diff
@@ -80,15 +80,8 @@ private String formatDuration(Duration duration) {
     return String.format("%d:%02d:%02d", absSeconds / 3600, (absSeconds % 3600) / 60, absSeconds % 60);
   }
 
-  private int countLines(File input) throws FileNotFoundException {
-    int count = 0;
-    try (Scanner sc = new Scanner(input)) {
-      while (sc.hasNextLine()) {
-        sc.nextLine();
-        count++;
-      }
-    }
-    return count;
+  private int countLines(File input) throws IOException {
+    return Files.readAllLines(input.toPath()).size();
```

Comment:
I'm not sure if this is a good idea, the input file could be very large and would be loaded into memory this way (the same is true for other cases in HttpApiSentenceChecler)...

Reply:
You are definitely right, this approach doesn't work for large files.

It has been changed, please let me know if other places also need lines to be read one-by-one.